aws/resource_aws_lightsail_load_balancer_test.go

```diff
@@ -0,0 +1,305 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSLightsailLoadBalancerConfig_basic(lightsailLoadBalancerName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAWSLightsailLoadBalancerExists("aws_lightsail_load_balancer.lightsail_load_balancer_test", &loadBalancer),
+					resource.TestCheckResourceAttrSet("aws_lightsail_load_balancer.lightsail_load_balancer_test", "health_check_path"),
+					resource.TestCheckResourceAttrSet("aws_lightsail_load_balancer.lightsail_load_balancer_test", "instance_port"),
+					resource.TestCheckResourceAttrSet("aws_lightsail_load_balancer.lightsail_load_balancer_test", "dns_name"),
+					resource.TestCheckResourceAttr("aws_lightsail_load_balancer.lightsail_load_balancer_test", "tags.%", "0"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSLightsailLoadBalancer_Name(t *testing.T) {
+	var conf lightsail.LoadBalancer
+	lightsailName := fmt.Sprintf("tf-test-lightsail-%d", acctest.RandInt())
+	lightsailNameWithSpaces := fmt.Sprint(lightsailName, "string with spaces")
+	lightsailNameWithStartingDigit := fmt.Sprintf("01-%s", lightsailName)
+	lightsailNameWithUnderscore := fmt.Sprintf("%s_123456", lightsailName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: "aws_lightsail_load_balancer.lightsail_load_balancer_test",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckAWSLightsailLoadBalancerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccAWSLightsailLoadBalancerConfig_basic(lightsailNameWithSpaces),
+				ExpectError: regexp.MustCompile(`must contain only alphanumeric characters, underscores, hyphens, and dots`),
+			},
+			{
+				Config:      testAccAWSLightsailLoadBalancerConfig_basic(lightsailNameWithStartingDigit),
+				ExpectError: regexp.MustCompile(`must begin with an alphabetic character`),
+			},
+			{
+				Config: testAccAWSLightsailLoadBalancerConfig_basic(lightsailName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAWSLightsailLoadBalancerExists("aws_lightsail_load_balancer.lightsail_load_balancer_test", &conf),
+					resource.TestCheckResourceAttrSet("aws_lightsail_load_balancer.lightsail_load_balancer_test", "health_check_path"),
+					resource.TestCheckResourceAttrSet("aws_lightsail_load_balancer.lightsail_load_balancer_test", "instance_port"),
+				),
+			},
+			{
+				Config: testAccAWSLightsailLoadBalancerConfig_basic(lightsailNameWithUnderscore),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAWSLightsailLoadBalancerExists("aws_lightsail_load_balancer.lightsail_load_balancer_test", &conf),
+					resource.TestCheckResourceAttrSet("aws_lightsail_load_balancer.lightsail_load_balancer_test", "health_check_path"),
+					resource.TestCheckResourceAttrSet("aws_lightsail_load_balancer.lightsail_load_balancer_test", "instance_port"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSLightsailLoadBalancer_Tags(t *testing.T) {
```

Comment:
see https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/contribution-checklists.md#resource-tagging-acceptance-testing-implementation

Reply:
Thanks for the feedback, this change will be implemented once the following pull request is merged by @akonrath https://github.com/akonrath/terraform-provider-aws/pull/96